packages/editor-ui/src/composables/useCanvasMapping.ts

```diff
@@ -87,6 +103,97 @@ export function useCanvasMapping({
 		}, {}),
 	);
 
+	const nodePinnedDataById = computed(() =>
+		workflow.value.nodes.reduce<Record<string, INodeExecutionData[] | undefined>>((acc, node) => {
+			acc[node.id] = workflowsStore.pinDataByNodeName(node.name);
+			return acc;
+		}, {}),
+	);
+
+	const nodeExecutionStatusById = computed(() =>
+		workflow.value.nodes.reduce<Record<string, ExecutionStatus>>((acc, node) => {
+			acc[node.id] =
+				workflowsStore.getWorkflowRunData?.[node.name]?.filter(Boolean)[0].executionStatus ?? 'new';
+			return acc;
+		}, {}),
+	);
+
+	const nodeExecutionRunDataById = computed(() =>
+		workflow.value.nodes.reduce<Record<string, ITaskData[] | null>>((acc, node) => {
+			acc[node.id] = workflowsStore.getWorkflowResultDataByNodeName(node.name);
+			return acc;
+		}, {}),
+	);
+
+	const nodeIssuesById = computed(() =>
+		workflow.value.nodes.reduce<Record<string, string[]>>((acc, node) => {
+			const issues: string[] = [];
+			const nodeExecutionRunData = workflowsStore.getWorkflowRunData?.[node.name];
+			if (nodeExecutionRunData) {
+				nodeExecutionRunData.forEach((executionRunData) => {
+					if (executionRunData?.error) {
+						const { message, description } = executionRunData.error;
+						const issue = `${message}${description ? ` (${description})` : ''}`;
+						issues.push(xss(issue));
```

Comment:
We already have a helper that wraps XSS functionality - `sanitizeHtml` couldn't we use it instead?

Reply:
Good point, updated!